### PR TITLE
SALTO-4577: DashboardGadget references

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add references to `Filter`, `Project` and extend `Field` references to support custom fields as well

---

_Additional context for reviewer_
This PR also fixes a bug in resolving of `Field` references in `DashboardGadget` causing deployment of changes with those references to fail

Will add noise reduction

---
_Release Notes_: 

_Jira_adapter_:
- Fix bug in deployment of DashboradGadget
- Add references to Filter, Project, and Field from DashboardGadget

---
_User Notifications_: 

_Jira_adapter_:
- References to Filter, Project, and Field will be added in DashboardGadget elements

### CODE REVIEW

**Review by netama — COMMENTED** [review]
the new reference rule and serialization LGTM 🎉 , leaving the other fixes for @alonstern :)

File: packages/jira-adapter/src/reference_mapping.ts [hunk 1]
```diff
@@ -91,8 +122,37 @@ const groupNameSerialize: GetLookupNameFunc = ({ ref }) =>
 const groupIdSerialize: GetLookupNameFunc = ({ ref }) =>
   (isInstanceElement(ref.value) ? ref.value.value.groupId : ref.value)
 
+const gadgetValueSerialize: GetLookupNameFunc = ({ ref, path, element }) => {
```
Comment (alonstern) [inline packages/jira-adapter/src/reference_mapping.ts:125]: Can you extract the gadget things to a separate file?

**Review by alonstern — APPROVED** [review]
LGTM 👌 I guess we need a noise reduction for this?

File: packages/jira-adapter/src/reference_mapping.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/jira-adapter/src/reference_mapping.ts:125]: sure

**Pull request merged**